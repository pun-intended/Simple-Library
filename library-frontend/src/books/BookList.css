.BookCard {
    max-height: 37vh;
    margin-bottom: 1vw;
}

.BookCover {
    min-height: 18vh;
    max-height: 18vh;
}

/* TODO - Fix text height to fit into cards */