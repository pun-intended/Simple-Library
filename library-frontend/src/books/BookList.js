// List all books available -> book card
// List all books checked out -> book card
// Click available book to show checkout card

import React, { useEffect, useState, useContext } from "react"
import BookCard from "./BookCard";
import LibraryApi from "../api";
import {Container, Col, Row} from 'reactstrap'
import StudentContext from "../StudentContext";
import "./BookList.css"

// TODO - Move books/update out, pass books as param to make {outstanding} {all books} {available} list
const BookList = () => {
    const {setStudents} = useContext(StudentContext)
    const [books, setBooks] = useState([])
    const [update, setUpdate] = useState(false)
    useEffect( () => {
        async function initializeList(){
            const updateBooks = await LibraryApi.getAllBooks()
            const updateStudents = await LibraryApi.getAllStudents()
            setBooks(updateBooks)
            setStudents(updateStudents)
            setUpdate(false)
        }
        initializeList()
    }, [update])

    return(
        <Container className="BookList">
            <h1>All Books</h1>
                <Row className="align-items-stretch"> 
                    {books.map((book) => {
                        return(
                            <BookCard book={book} setUpdate={setUpdate} key={book.id}/> 
                        )
                    })}
                </Row>     
        </Container>
    )
}

export default BookList;