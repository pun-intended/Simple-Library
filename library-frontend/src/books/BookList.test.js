import React from "react";
import { render } from "@testing-library/react";
import BookList from "./BookList";
import StudentContext from "../StudentContext";


const books = {
    books:[{
        id: "101",
        title: "test book",
        isbn: "999-999-9999",
        stage: 2,
        condition: "great",
        available: "good",
    },
    {
        id: "102",
        title: "test book",
        isbn: "999-999-9999",
        stage: 2,
        condition: "great",
        available: "good",
    },
    {
        id: "102",
        title: "test book",
        isbn: "999-999-9999",
        stage: 2,
        condition: "great",
        available: "good",
    }]
};
const students = {
    students: [{
        fist_name: "test",
        last_name: "student",
    },
    {
        fist_name: "test",
        last_name: "student",
        book_id: "101",
    }]
};

// Smoke test
it("renders without crashing", () => {
    render(<StudentContext.Provider value={students}><BookList /></StudentContext.Provider>)
});

// Snapshot
it("matches snapshot", () => {
    const {asFragment} = render(<StudentContext.Provider value={students}><BookList /></StudentContext.Provider>);
    expect(asFragment()).toMatchSnapshot();
});

// --- Need to mock LibraryApi to test  ---
// // renders all books from test data
// it("renders all books from test data", () => {
//     const {getAllByText} = render(<StudentContext.Provider value={students}><BookList /></StudentContext.Provider>);
//     books.books.forEach(book => {
//         expect(getAllByText(book.title)).toBeTruthy();
//     });
// });