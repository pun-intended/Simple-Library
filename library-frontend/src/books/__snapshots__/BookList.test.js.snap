// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BookList container"
  >
    <h1>
      All Books
    </h1>
    <div
      class="card-group"
    />
  </div>
</DocumentFragment>
`;
